torch/csrc/multiprocessing/init.cpp

```diff
@@ -38,6 +39,15 @@ PyObject* multiprocessing_init(PyObject* _unused, PyObject* noargs) {
   Py_RETURN_TRUE;
 }
 
+PyObject* set_thread_name(PyObject* _unused, PyObject* arg) {
+  TORCH_CHECK(THPUtils_checkString(arg), "invalid argument to setDevice");
+
+  auto name = THPUtils_unpackString(arg);
```

Comment:
Just a question:
I'm guessing it's ok for a user to set thread name to same name for multiple threads? We don't need to remember/warn or protect the user from doing something erroneous?

Reply:
Also, do we care if the string is large? Or restrict to a fixed size?

Reply:
Currently we all threads to `pt_main_thread` by default -- so that's definitely fine. Length wise, I think there is a max but it just silently gets truncated by pthreads so hasn't been ever an issue from what I've seen.

Reply:
The max is 15 characters + null -- setThreadName truncates it automatically